site-permissions/site-permissions-api/src/main/java/com/duckduckgo/site/permissions/api/SitePermissionsManager.kt

```diff
@@ -20,6 +20,17 @@ import android.webkit.PermissionRequest
 
 /** Public interface for managing site permissions data */
 interface SitePermissionsManager {
+    fun isPermissionAllowedToAsk(
+        url: String,
+        permission: String
+    ): Boolean
+
+    fun isPermissionGranted(
+        url: String,
+        tabId: String,
+        permission: String
+    ): Boolean
```

Comment:
In the Android team we have a process for adding new public API surface. More so give this is contribution outside the functional team. Please check [this task](https://app.asana.com/0/1202561462274611/1205060262189411/f)

Reply:
Yup, it would be great to discuss what are the specific needs to figure out what's the best option . 

For example, I get the difference here is that we don't get a PermissionRequest but directly a string with the actual permission, it is however not clear to me if we can get more than one permission on the same message. I'm assuming the answer is yes which then comes back to the c-s-s implementation of this but the PR is not the place to discuss that.

Reply:
Sounds good. Reworked this based on the update codebase and submitted change proposal: https://app.asana.com/0/1202552961248957/1205981799209335/f. Once we settle on it, I can add more tests for PermissionsManager.